rust/main/agents/relayer/src/relayer/origin.rs

```diff
@@ -0,0 +1,361 @@
+                db,
+                self.advanced_log_meta,
+                self.tx_id_indexing_enabled,
+            )
+            .await
+            .map(|r| r as Arc<dyn ContractSyncer<_>>)
+            .map_err(|err| FactoryError::MessageSync(domain.to_string(), err.to_string())),
+        }
+    }
+
+    async fn init_igp_sync(
+        &self,
+        domain: &HyperlaneDomain,
+        chain_conf: &ChainConf,
+        db: Arc<HyperlaneRocksDB>,
+    ) -> Result<InterchainGasPaymentSync, FactoryError> {
+        match InterchainGasPayment::indexing_cursor(domain.domain_protocol()) {
+            CursorType::SequenceAware => Self::build_sequenced_contract_sync(
+                domain,
+                chain_conf,
+                &self.core_metrics,
+                &self.sync_metrics,
+                db,
+                self.advanced_log_meta,
+                false,
+            )
+            .await
+            .map(|r| r as Arc<dyn ContractSyncer<_>>)
+            .map_err(|err| {
+                FactoryError::InterchainGasPaymentSync(domain.to_string(), err.to_string())
+            }),
+            CursorType::RateLimited => Self::watermark_contract_sync(
+                domain,
+                chain_conf,
+                &self.core_metrics,
+                &self.sync_metrics,
+                db,
+                self.advanced_log_meta,
+                false,
+            )
+            .await
+            .map(|r| r as Arc<dyn ContractSyncer<_>>)
+            .map_err(|err| {
+                FactoryError::InterchainGasPaymentSync(domain.to_string(), err.to_string())
+            }),
+        }
+    }
+
+    async fn init_merkle_tree_hook_sync(
+        &self,
+        domain: &HyperlaneDomain,
+        chain_conf: &ChainConf,
+        db: Arc<HyperlaneRocksDB>,
+    ) -> Result<MerkleTreeHookSync, FactoryError> {
+        match MerkleTreeInsertion::indexing_cursor(domain.domain_protocol()) {
+            CursorType::SequenceAware => Self::build_sequenced_contract_sync(
+                domain,
+                chain_conf,
+                &self.core_metrics,
+                &self.sync_metrics,
+                db,
+                self.advanced_log_meta,
+                false,
+            )
+            .await
+            .map(|r| r as Arc<dyn ContractSyncer<_>>)
+            .map_err(|err| FactoryError::MerkleTreeHookSync(domain.to_string(), err.to_string())),
+            CursorType::RateLimited => Self::watermark_contract_sync(
+                domain,
+                chain_conf,
+                &self.core_metrics,
+                &self.sync_metrics,
+                db,
+                self.advanced_log_meta,
+                false,
+            )
+            .await
+            .map(|r| r as Arc<dyn ContractSyncer<_>>)
+            .map_err(|err| FactoryError::MerkleTreeHookSync(domain.to_string(), err.to_string())),
+        }
+    }
+
+    pub async fn build_sequenced_contract_sync<T, S>(
```

Comment:
This does not need to be `pub`, I guess.

Reply:
Fixed